airflow/api_connexion/endpoints/xcom_endpoint.py

```diff
@@ -14,25 +14,47 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-from flask import request
+from typing import List, Optional
+
+from flask import Response, request
+from marshmallow import ValidationError
 from sqlalchemy import and_, func
 from sqlalchemy.orm.session import Session
 
 from airflow.api_connexion import parameters
-from airflow.api_connexion.exceptions import NotFound
+from airflow.api_connexion.exceptions import BadRequest, NotFound
 from airflow.api_connexion.schemas.xcom_schema import (
     XComCollection, XComCollectionItemSchema, XComCollectionSchema, xcom_collection_item_schema,
-    xcom_collection_schema,
+    xcom_collection_schema, xcom_schema,
 )
 from airflow.models import DagRun as DR, XCom
 from airflow.utils.session import provide_session
 
 
-def delete_xcom_entry():
+@provide_session
+def delete_xcom_entry(
+    dag_id: str,
+    dag_run_id: str,
+    task_id: str,
+    xcom_key: str,
+    session: Session
+) -> Response:
```

Comment:
```suggestion
):
```
Mypy doesn't raise error it gets lost in [types](https://github.com/apache/airflow/pull/9329#discussion_r443198113)